### ISSUE TITLE
Ajax request to get older logs in AdminActivityLogPage cannot complete when the number of logs is large

### ISSUE THREAD

**Issue by bamboo3250** [TEAMMATES/repo#4027]
![screen shot 2015-07-27 at 2 40 02 pm](https://cloud.githubusercontent.com/assets/8078323/8899987/843af14e-346d-11e5-969e-66a3f76bc105.png)

It keeps loading....

![screen shot 2015-07-27 at 2 40 07 pm](https://cloud.githubusercontent.com/assets/8078323/8899989/86885e1e-346d-11e5-8ea7-e2fa7936f206.png)

**Comment by damithc** [comment]
I did  not have this issue with the live server. Under what circumstances does this happen?

**Comment by bamboo3250** [comment]
It happens to me when I have 50 logs or above so that the "Older Logs" appears then I click on it. It just keeps showing the loading icon.

I made some investigation and went to this page https://code.google.com/p/googleappengine/issues/detail?id=8987

**Comment by bamboo3250** [comment]
This is the exception it threw

```
Jul 27, 2015 8:30:18 AM teammates.ui.controller.ControllerServlet doPost
SEVERE: TEAMMATESLOG|||admin|||System Error Report|||true|||Unknown|||Unknown|||Unknown|||Unknown|||<span class="text-danger">TEAMMATES (4.18): New System Exception: java.lang.NumberFormatException: For input string: "55b5ec040004f1a0da4b9237"</span><br><a href="#" onclick="showHideErrorMessage('error1139064774');">Show/Hide Details >></a><br><span id="error1139064774" style="display: none;"><b>Error Message</b>
<br/>
<pre><code>
java.lang.NumberFormatException: For input string: "55b5ec040004f1a0da4b9237"
</code></pre>
<br/>

<b>Request Path</b>
/admin
<br/>

<b>Request Parameters</b>
{filterQuery::, offset::Chg1NWI1ZWMwNDAwMDRmMWEwZGE0YjkyMzc=, all::false, testdata::false}
<br/>

<b>Stack Trace</b>
<pre><code>

com.google.appengine.api.log.LogServiceException: java.lang.NumberFormatException: For input string: "55b5ec040004f1a0da4b9237"
	at com.google.appengine.api.log.LogServiceImpl.fetch(LogServiceImpl.java:38)
	at com.google.appengine.api.log.LogServiceImpl.fetch(LogServiceImpl.java:24)
	at teammates.ui.controller.AdminActivityLogPageAction.getAppLogs(AdminActivityLogPageAction.java:180)
	at teammates.ui.controller.AdminActivityLogPageAction.execute(AdminActivityLogPageAction.java:88)
	at teammates.ui.controller.Action.executeAndPostProcess(Action.java:335)
	at teammates.ui.controller.ControllerServlet.doPost(ControllerServlet.java:58)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:637)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
	at teammates.storage.datastore.DatastoreFilter.doFilter(DatastoreFilter.java:28)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.api.socket.dev.DevSocketFilter.doFilter(DevSocketFilter.java:74)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.development.ResponseRewriterFilter.doFilter(ResponseRewriterFilter.java:127)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.development.HeaderVerificationFilter.doFilter(HeaderVerificationFilter.java:34)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.api.blobstore.dev.ServeBlobFilter.doFilter(ServeBlobFilter.java:63)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:43)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.development.StaticFileFilter.doFilter(StaticFileFilter.java:125)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.development.DevAppServerModulesFilter.doDirectRequest(DevAppServerModulesFilter.java:366)
	at com.google.appengine.tools.development.DevAppServerModulesFilter.doDirectModuleRequest(DevAppServerModulesFilter.java:349)
	at com.google.appengine.tools.development.DevAppServerModulesFilter.doFilter(DevAppServerModulesFilter.java:116)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
	at com.google.appengine.tools.development.DevAppEngineWebAppContext.handle(DevAppEngineWebAppContext.java:98)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at com.google.appengine.tools.development.JettyContainerService$ApiProxyHandler.handle(JettyContainerService.java:491)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
	at org.mortbay.jetty.HttpParser.parseNext(HttpParser.java:547)
	at org.mortbay.jetty.HttpParser.parseAvailable(HttpParser.java:212)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at org.mortbay.io.nio.SelectChannelEndPoint.run(SelectChannelEndPoint.java:409)
	at org.mortbay.thread.QueuedThreadPool$PoolThread.run(QueuedThreadPool.java:582)

</code></pre>


</span>|||/admin/adminActivityLogPage?offset=Chg1NWI1ZWMwNDAwMDRmMWEwZGE0YjkyMzc%3D&all=false&testdata=false&filterQuery=

```

**Comment by bamboo3250** [comment]
It seems that it cannot retrieve a certain number of logs from the LogService

![screen shot 2015-07-27 at 4 50 27 pm](https://cloud.githubusercontent.com/assets/8078323/8901958/baf0e8bc-347f-11e5-91f4-214cacaa3bde.png)

**Comment by wkurniawan07** [comment]
This is fixed in SDK 1.9.19, as mentioned in [the release notes](https://code.google.com/p/googleappengine/wiki/SdkForJavaReleaseNotes). Upgrading really does a lot.
